openwisp_radius/api/views.py

```diff
@@ -678,3 +675,39 @@ def create_phone_token(self, *args, **kwargs):
 
 
 change_phone_number = ChangePhoneNumberView.as_view()
+
+
+class IsAllowedRadiusBatch(BasePermission):
+    def has_permission(self, request, view):
+        radbatch_model_name = swapper.get_model_name('openwisp_radius', 'RadiusBatch')
+        app_label, _ = swapper.split(radbatch_model_name)
+        return bool(
+            request.user
+            and (request.user.is_superuser or request.user.is_staff)
+            and request.user.has_perm(f'{app_label}.view_radiusbatch')
+        )
+
+
+class DownloadRadiusBatchPdfView(DispatchOrgMixin, APIView):
+    authentication_classes = (UserTokenAuthentication, SessionAuthentication)
+    permission_classes = (IsAllowedRadiusBatch,)
+
+    def get(self, request, *args, **kwargs):
+        # TODO:N if a superuser is not in org, should they be
+        # allowed? because they are superuser
+        # and can basically grant permission to themselves!
```

Comment:
Quick question. What do you think about this change.
The old `validate_membership` doesn't even allow super to access without the superuser being part of the organization.
This is unexpected, as a superuser can basically give themselves the permission, so, why are they rejected?

Reply:
I am not sure if it was on purpose or not, but supporting superusers was not a priority because these endpoints are used mostly by non superusers.

Does any test fail if we change it to take superusers into account?

Reply:
No, none of tests failed, 
Hence, I've added in the `validate_membership` check:
```
if not (user.is_superuser or (self.organization.pk,) in user.organizations_pk):
  .. 
  raise..
```

Let me know if I should remove it or let it be! :smile: